spring-core/src/main/java/org/springframework/core/MethodParameter.java

```diff
@@ -647,7 +647,7 @@ public Annotation[] getParameterAnnotations() {
 				// for inner classes, so access it with the actual parameter index lowered by 1
 				index = this.parameterIndex - 1;
 			}
-			paramAnns = (index >= 0 && index < annotationArray.length ?
+			paramAnns = (index >= 0 && index < annotationArray.length && annotationArray[index].length > 0 ?
```

Comment:
This optimization introduces a small change of behavior, but it looks acceptable to me as adaptation of an empty array is IMO unlikely to produce something else than an empty array.